src/PathautoManager.php

```diff
@@ -218,31 +235,31 @@ public function createAlias($module, $op, $source, $data, $type = NULL, $langcod
   /**
    * {@inheritdoc}
    */
-  public function getPatternByEntity($entity_type_id, $bundle = '', $language = LanguageInterface::LANGCODE_NOT_SPECIFIED) {
-    $config = $this->configFactory->get('pathauto.pattern');
-
-    $pattern_id = "$entity_type_id:$bundle:$language";
-    if (!isset($this->patterns[$pattern_id])) {
-      $pattern = '';
-      $variables = array();
-      if ($language != LanguageInterface::LANGCODE_NOT_SPECIFIED) {
-        $variables[] = "{$entity_type_id}.bundles.{$bundle}.languages.{$language}";
-      }
-      if ($bundle) {
-        $variables[] = "{$entity_type_id}.bundles.{$bundle}.default";
-      }
-      $variables[] = "{$entity_type_id}.default";
-
-      foreach ($variables as $variable) {
-        if ($pattern = trim($config->get('patterns.' . $variable))) {
-          break;
+  public function getPatternByEntity(EntityInterface $entity) {
+    if (!isset($this->patterns[$entity->getEntityTypeId()][$entity->id()])) {
+      foreach (\Drupal::service('plugin.manager.alias_type')->getPluginDefinitionByType($entity->getEntityTypeId()) as $plugin_id => $plugin_definition) {
+        /** @var \Drupal\pathauto\PathautoPatternInterface[] $patterns */
+        $patterns = \Drupal::entityManager()
+          ->getStorage('pathauto_pattern')
+          ->loadByProperties(['type' => $plugin_id]);
```

Comment:
We should add 
 *   lookup_keys = {
 *     "type"
 *   }

to the annotation, then drupal automatically manages an index for this in key value store.